File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int> map;
        for(int num:nums){
            map[num]++;
            if(map[num]>n/2){
                return num;
            }
        }
        return 0;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }
        List<Integer>[] bucket = new List[nums.length+1];
        for(int num:map.keySet()){
            int freq=map.get(num);
            if(bucket[freq]==null){
                bucket[freq]=new ArrayList<>();
            }
            bucket[freq].add(num);
        }
        int[] ans=new int[k];
        int idx=0;
        for(int i=bucket.length-1;i>=0 && idx<k;i--){
            if(bucket[i]!=null){
                for(int n:bucket[i]){
                    ans[idx++]=n;
                    if(idx==k) break;
                }
            }
        }
        return ans;
    }
}
